drivers/scheduler/anthos/anthos.go

```diff
@@ -643,6 +666,88 @@ func (anth *anthos) updateFileOwnership(dirPath string) error {
 	return nil
 }
 
+// dumpUpgradeLogs collects upgrade logs
+func (anth *anthos) dumpUpgradeLogs(clusterName string) error {
+	adminKubeConfPath := path.Join(anth.confPath, kubeConfig)
+	adminInstance, err := core.NewInstanceFromConfigFile(adminKubeConfPath)
+	if err != nil {
+		return fmt.Errorf("creating admin cluster instance failing with error. Err: (%v)", err)
+	}
+	if err := anth.collectUpgradeLogsInNameSpace(adminInstance, kubeSystemNameSpace); err != nil {
+		return err
+	}
+	// Collecting pods logs from usercluster namespace
+	if err := anth.collectUpgradeLogsInNameSpace(adminInstance, clusterName); err != nil {
+		return err
+	}
+	return nil
+}
+
+// collectUpgradeLogsInNameSpace collect upgrade logs for namespace provided
+func (anth *anthos) collectUpgradeLogsInNameSpace(k8sInstance k8s.Ops, namespace string) error {
+	clusterApiLabel := make(map[string]string, 0)
+	clusterApiLabel[clusterApiKey] = clusterApiValue
+	podList, err := k8sInstance.GetPods(namespace, clusterApiLabel)
+	if err != nil {
+		return fmt.Errorf("retrieving cluster api pod is failing with error. Err: (%v)", err)
+	}
+	// Collecting pods logs from kube-system namespace
```

Comment:
Also check `len(pod.Items) > 0` and throw appropriate error. If this check is not present, then we will write empty logs as this function will return without any error for 0 pods as well.

Reply:
Incorporated !!